pipelines/api-pipelines.yml

```diff
@@ -20,137 +22,57 @@ pr:
-
-          - task: DotNetCoreCLI@2
-            displayName: Unit Tests
-            inputs:
-              command: test
-              projects: '**/LinkyLink.Tests/*.csproj'
-              arguments: -c $(BuildConfiguration) --collect:"XPlat Code Coverage"
-
-            # Generate the report using ReportGenerator (https://github.com/danielpalme/ReportGenerator)
-          - script: |
-              dotnet tool install -g dotnet-reportgenerator-globaltool
-              reportgenerator -reports:$(Agent.TempDirectory)/**/coverage.cobertura.xml -targetdir:$(Build.SourcesDirectory)/coverlet/reports -reporttypes:"Cobertura"
-            displayName: Create Code coverage report
-
-          - task: PublishCodeCoverageResults@1
-            displayName: 'Publish code coverage'
-            inputs:
-              codeCoverageTool: Cobertura
-              summaryFileLocation: $(Build.SourcesDirectory)/coverlet/reports/Cobertura.xml 
-              failIfCoverageEmpty: true 
+      - template: templates/api-build.yml
 
-          - task: BuildQualityChecks@6
-            displayName: 'Build quality check'
-            inputs:
-              checkCoverage: true
-              coverageFailOption: 'fixed'
-              coverageType: 'lines'
-              coverageThreshold: '25'
-
-          - task: DotNetCoreCLI@2
-            displayName: Publish packages
-            inputs:
-              command: publish
-              arguments: '--configuration $(BuildConfiguration) --output $(Build.ArtifactStagingDirectory)'
-              zipAfterPublish: True
-
-          - task: PublishBuildArtifacts@1
-            condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
-            displayName: Publish Artifacts
-            inputs:
-              pathtoPublish: '$(Build.ArtifactStagingDirectory)' 
-              artifactName: 'drop'
-              publishLocation: 'Container'
-
-  - stage: DeployStage
-    dependsOn: BuildStage
+  - stage: Development
+    displayName: Development
+    dependsOn: BuildAndValidate
     condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
-    displayName: Deploy
+    variables:
+      - group: prodoh-urlist-dev
     jobs:
-      - job: RegionalMaxtrixDeployment
-        displayName: API to Region
-        strategy:
-          matrix:
-            westus2:
-              region: westus2
-            eastus:
-              region: eastus
-            centralus:
-              region: centralus
-        steps:
-          - task: UseDotNet@2
-            displayName: Dotnet version 3.1.100
-            inputs:
-              version: '3.1.100'
+      - template: templates/api-deploy-environment.yml
+        parameters:
+          subscription: "PELITTLE TEAM - CSE DWR(d36d0808-a967-4f73-9fdc-32ea232fc81d)"
+          businessUnit: $(businessUnit)
+          appName: $(appName)
+          environment: dev
+          regions:
+            - westus2
```

Comment:
I suspect the `region` should be a configuration parameter?  (Similar comments below)

Reply:
yeah, we can do that.